magefiles/helm.go

```diff
@@ -0,0 +1,110 @@
+//go:build mage_helm
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/magefile/mage/sh"
+	"golang.org/x/xerrors"
+	"gopkg.in/yaml.v3"
+)
+
+const chartFile = "./helm/trivy/Chart.yaml"
+
+func main() {
+	trivyVersion, err := version()
+	if err != nil {
+		log.Fatalf("could not determine Trivy version: %v", err)
+	}
+	input, err := os.ReadFile(chartFile)
+	if err != nil {
+		log.Fatalf("could not find helm chart %s: %v", chartFile, err)
+	}
+	jsonData := map[string]interface{}{}
+	if err := yaml.Unmarshal(input, &jsonData); err != nil {
+		log.Fatalf("could not unmarshal helm chart %s: %v", chartFile, err)
+	}
+	currentAppVersion, ok := jsonData["appVersion"].(string)
+	if !ok {
+		log.Fatalf("could not determine current app version")
+	}
+	currentHelmVersion, ok := jsonData["version"].(string)
+	if !ok {
+		log.Fatalf("could not determine current helm version")
+	}
+	newHelmVersion := newHelmVersion(currentHelmVersion, currentAppVersion, trivyVersion)
+
+	log.Printf("Current helm version %q with Trivy %q will bump up %q with Trivy %q",
+		currentHelmVersion, currentAppVersion, newHelmVersion, trivyVersion)
+
+	newBranch := fmt.Sprintf("ci/helm-chart/bump-trivy-to-%s", trivyVersion)
+	title := fmt.Sprintf("ci(helm): bump Trivy version to %s", trivyVersion)
+	description := fmt.Sprintf("This PR bumps Trivy up to the %s version for the Helm chart.", trivyVersion)
+
+	cmds := [][]string{
+		[]string{"sed", "-i", "-e", fmt.Sprintf("s/appVersion: %s/appVersion: %s/g", currentAppVersion, trivyVersion), chartFile},
+		[]string{"sed", "-i", "-e", fmt.Sprintf("s/version: %s/version: %s/g", currentHelmVersion, trivyVersion), chartFile},
+		[]string{"git", "switch", "-c", newBranch},
+		[]string{"git", "add", "./helm/trivy/Chart.yaml"},
+		[]string{"git", "commit", "-m", title},
+		[]string{"git", "push", "origin", newBranch},
+		[]string{"gh", "pr", "create", "--base", "main", "--head", newBranch, "--title", title, "--body", description, "--repo", "$GITHUB_REPOSITORY"},
+	}
+
+	if err := runShCommands(cmds); err != nil {
+		log.Fatal(err)
+	}
+	log.Print("Successfully created PR with a new helm version")
+}
+
+func runShCommands(cmds [][]string) error {
+	for _, cmd := range cmds {
+		if err := sh.Run(cmd[0], cmd[1:]...); err != nil {
+			return xerrors.Errorf("failed to run %v: %w", cmd, err)
+		}
+	}
+	return nil
+}
+
+func splitVersion(version string) []int {
+	items := strings.Split(version, ".")
+	result := make([]int, len(items))
+	for i, item := range items {
+		result[i], _ = strconv.Atoi(item)
+	}
+	return result
+}
+
+func newHelmVersion(currentHelm, currentTrivy, newTrivy string) string {
```

Comment:
What do you think about using https://github.com/aquasecurity/go-version/blob/main/pkg/semver/version.go?
you can get [minor and patch parts](https://github.com/aquasecurity/go-version/blob/cf8a8d29271d2ecb1495127879994507b75099f0/pkg/semver/version.go#L172-L180)
also you can [increment version part](https://github.com/aquasecurity/go-version/blob/cf8a8d29271d2ecb1495127879994507b75099f0/pkg/semver/version.go#L122-L139)

Reply:
done https://github.com/aquasecurity/trivy/pull/7526/commits/1e4b16cb756eaaff325c0fa90e81283d81300a32